ClientApp/viking-vault/src/UserDashboard/CreateAutomaticPaymentModal.tsx

```diff
@@ -0,0 +1,255 @@
+            },
+            valid: null,
+            errorLabel: "",
+            companies: [],
+        }
+    }
+
+    private handleChange = (inputValue: string, inputName: string) => {
+        this.setState({
+            automaticPayment: {
+                ...this.state.automaticPayment,
+                [inputName]: inputValue
+            }
+        });
+    }
+
+    private isCompanySelected = () => {
+        if(this.state.automaticPayment.companyName !== "")
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "A company must be selected!",
+            class: "alert alert-info"
+        }
+    }
+
+    private amountMustBePositive = () => {
+        if(this.state.automaticPayment.amount.match(regexCheckIfPositiveFloat))
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "The amount must be a positive number!",
+            class: "alert alert-info"
+        }
+    }
+
+    private initialPaymentDateIsAfterCurrentDate = () => {
+        if(this.state.automaticPayment.initialPaymentDate !== "" && new Date(this.state.automaticPayment.initialPaymentDate) >= currentDate)
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "Initial Payment Date must be after the current date!",
+            class: "alert alert-info"
+        }
+    }
+
+    private mandatoryFieldsCompletedCorrectly = () => {
+        let val = false;
+        const validationFunctions = [this.isCompanySelected, this.amountMustBePositive, this.initialPaymentDateIsAfterCurrentDate];
+        val = validationFunctions.every(functionItem => functionItem().message === "Ok");
+        return val;
+    }
+
+    private resetInputValues = () => {
+        this.setState({
+            automaticPayment: {
+                ...this.state.automaticPayment,
+                companyName: "",
+                amount: "",
+                initialPaymentDate: "",
+            }
+        });
+    }
+
+    private closeModal = () => {
+        this.resetInputValues();
+        this.props.onModalClose(false);
+    }
+
+    private getIdOfCompanyByName = (name: string) => {
+        return this.state.companies.filter(company => company.name === name)[0].id;
+    }
+
+    private getAutomaticPayment = () => {
+        return {
+            companyId: this.getIdOfCompanyByName(this.state.automaticPayment.companyName),
+            companyName: this.state.automaticPayment.companyName,
+            amount: this.state.automaticPayment.amount,
+            initialPaymentDate: new Date(this.state.automaticPayment.initialPaymentDate),
+            lastPaymentDate: new Date(2000,0,2),
+        };
+    }
+
+    getAllCompanies = () => {
+        var companies;
```

Comment:
don't use var. Use let or const. let for variables that can be reassigned and const for read only variables